--- /usr/lib64/ruby/gems/2.1.0/gems/fog-1.25.0/lib/fog/openstack/requests/network/create_subnet.rb.org  2014-11-24 13:13:21.003002011 +0000
+++ /usr/lib64/ruby/gems/2.1.0/gems/fog-1.25.0/lib/fog/openstack/requests/network/create_subnet.rb      2014-11-24 13:12:09.965001030 +0000
@@ -15,7 +15,12 @@
                              :dns_nameservers, :host_routes, :enable_dhcp,
                              :tenant_id]
           vanilla_options.reject{ |o| options[o].nil? }.each do |key|
-            data['subnet'][key] = options[key]
+            if  options[:gateway_ip] == 'None' then
+              data['subnet'][key] = options[key]
+              data['subnet'][:gateway_ip] = nil
+            else
+              data['subnet'][key] = options[key]
+            end
           end

           request(

